hudi-spark-datasource/hudi-spark2/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/Spark24HoodieParquetReader.scala

```diff
@@ -0,0 +1,222 @@
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.avro.AvroDeserializer
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.codegen.GenerateUnsafeProjection
+import org.apache.spark.sql.catalyst.expressions.{Cast, JoinedRow, UnsafeRow}
+import org.apache.spark.sql.catalyst.util.DateTimeUtils
+import org.apache.spark.sql.execution.datasources.{PartitionedFile, RecordReaderIterator}
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.sources.Filter
+import org.apache.spark.sql.types.{AtomicType, StructField, StructType}
+import org.apache.spark.util.SerializableConfiguration
+
+import java.net.URI
+
+object Spark24HoodieParquetReader {
+
+  /**
+   * Get properties needed to read a parquet file
+   *
+   * @param vectorized true if vectorized reading is not prohibited due to schema, reading mode, etc
+   * @param sqlConf    the [[SQLConf]] used for the read
+   * @param options    passed as a param to the file format
+   * @param hadoopConf some configs will be set for the hadoopConf
+   * @return map of properties needed for reading a parquet file
+   */
+  def getPropsForReadingParquet(vectorized: Boolean,
+                                sqlConf: SQLConf,
+                                options: Map[String, String],
+                                hadoopConf: Configuration): Map[String, String] = {
+    //set hadoopconf
+    hadoopConf.set(ParquetInputFormat.READ_SUPPORT_CLASS, classOf[ParquetReadSupport].getName)
+    hadoopConf.set(SQLConf.SESSION_LOCAL_TIMEZONE.key, sqlConf.sessionLocalTimeZone)
+    hadoopConf.setBoolean(SQLConf.CASE_SENSITIVE.key, sqlConf.caseSensitiveAnalysis)
+    hadoopConf.setBoolean(SQLConf.PARQUET_BINARY_AS_STRING.key, sqlConf.isParquetBinaryAsString)
+    hadoopConf.setBoolean(SQLConf.PARQUET_INT96_AS_TIMESTAMP.key, sqlConf.isParquetINT96AsTimestamp)
+
+    Map(
+      "enableVectorizedReader" -> vectorized.toString,
+      "enableParquetFilterPushDown" -> sqlConf.parquetFilterPushDown.toString,
+      "pushDownDate" -> sqlConf.parquetFilterPushDownDate.toString,
+      "pushDownTimestamp" -> sqlConf.parquetFilterPushDownTimestamp.toString,
+      "pushDownDecimal" -> sqlConf.parquetFilterPushDownDecimal.toString,
+      "pushDownInFilterThreshold" -> sqlConf.parquetFilterPushDownInFilterThreshold.toString,
+      "pushDownStringStartWith" -> sqlConf.parquetFilterPushDownStringStartWith.toString,
+      "isCaseSensitive" -> sqlConf.caseSensitiveAnalysis.toString,
+      "timestampConversion" -> sqlConf.isParquetINT96TimestampConversion.toString,
+      "enableOffHeapColumnVector" -> sqlConf.offHeapColumnVectorEnabled.toString,
+      "capacity" -> sqlConf.parquetVectorizedReaderBatchSize.toString,
+      "returningBatch" -> sqlConf.parquetVectorizedReaderEnabled.toString,
+      "enableRecordFilter" -> sqlConf.parquetRecordFilterEnabled.toString,
+      "timeZoneId" -> sqlConf.sessionLocalTimeZone
```

Comment:
If these keys are for internal use in Hudi, let's define them in a common place for reuse across multiple Spark versions.

Reply:
The keys are the name of the variables that spark uses

Reply:
no longer relevant